refinery/analysis_manager/tests.py

```diff
@@ -185,108 +188,12 @@ class AnalysisViewsTests(AnalysisManagerTestBase, ToolManagerTestBase):
     def setUp(self):
         # super() will only ever resolve a single class type for a given method
         AnalysisManagerTestBase.setUp(self)
-        ToolManagerTestBase.setUp(self)
```

Comment:
I think its odd to have the setUp down below. I think the only change needed here would be to assign the proper perms to the requesting user while leaving `ToolManagerTestBase.setUp(self)` where it was:
``` python
assign_perm(
    'core.read_meta_dataset',
    self.user,
    self.analysis.data_set
)
```

Reply:
The issue is AnalysisManagerTestBase and ToolManagerTestBase both assign a self.user.
It's not a perms issue. In the walk through, (please feel free to confirm) the incorrect user is assign for the Non-tool unit test. Note the non-tool unit test does not need the toolmanagertestbase set up. These unit tests should be refactored and cleaned up. It was really difficult to debug these test because of the conflicting scope with multiple class set ups.

Reply:
Looking again I think it would be fine to refactor `AnalysisViewsTests` to:
```
class AnalysisViewsTests(ToolManagerTestBase):
    """
    Tests for `analysis_manager.views`
    """
    def setUp(self):
        super(AnalysisViewsTests, self).setUp()
        self.request_factory = RequestFactory()

    @mock.patch.object(AnalysisStatus,
                       "galaxy_file_import_state",
                       return_value="coffee")
    def test_tool_analysis_calls_galaxy_file_import_state(
            self,
            file_import_state_mock
    ):
        self.create_tool(ToolDefinition.WORKFLOW)
        self.status_url_root = "/analysis_manager/{}/".format(
            self.tool.analysis.uuid
        )
        request = self.request_factory.get(
            self.status_url_root,
            content_type="application/json"
        )
        request.user = self.user
        response = analysis_status(request, self.tool.analysis.uuid)
        self.assertTrue(file_import_state_mock.called)

        self.assertEqual(
            json.loads(response.content),
            {
                "galaxyAnalysis": [],
                "refineryImport": [],
                "galaxyImport": "coffee",
                "overall": "INITIALIZED",
                "galaxyExport": []
            }
        )
``` 

There was a time where we had a split between Tool & non-Tool analyses and since we only deal with Tool-based Analyses nowadays I think we could get rid of the other test

Reply:
I would avoid inheritance in test cases completely. Tests don't need to follow DRY.